File: examples/CBuildr6/ExNbkU.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExNbkU.h"
//---------------------------------------------------------------------------
#pragma link "OvcSC"
#pragma link "OvcBase"
#pragma link "OvcNbk"
#pragma link "OvcSF"
#pragma link "OvcEF"
#pragma link "OvcPF"
#pragma link "OvcPB"
#pragma link "OvcNF"
#pragma link "OvcCal"
#pragma link "OvcTimer"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnTabPosClick(TObject *Sender)
{
  if (NB1->TabOrientation == toRight)
    NB1->TabOrientation = toTop;
  else
    NB1->TabOrientation = toRight;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnShadowTextClick(TObject *Sender)
{
  NB1->ShadowedText = !NB1->ShadowedText;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnChangeTabClick(TObject *Sender)
{
  NB1->PageIndex = OvcSimpleField1->AsInteger;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnFontColorClick(TObject *Sender)
{
  ColorDialog1->Color = NB1->Font->Color;
  ColorDialog1->Execute();
  NB1->Font->Color = ColorDialog1->Color;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnFontSelClick(TObject *Sender)
{
  FontDialog1->Font = NB1->Font;
  FontDialog1->Execute();
  NB1->Font = FontDialog1->Font;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnCalDateClick(TObject *Sender)
{
  // second page (index 1) contains the calendar
  NB1->PageIndex = 1;
  // change the calendar date
  OvcCalendar1->CalendarDate = OvcPictureField1->AsOvcDate;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::OvcCalendar1Change(TObject *Sender, TDateTime Date)
{
  String S = (int)OvcCalendar1->Year;
  Panel1->Caption = OvcCalendar1->DateString("www, dd nnn yyyy");
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnTabDisableClick(TObject *Sender)
{
  NB1->Pages[3]->Enabled = !NB1->Pages[3]->Enabled;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
  char Buf[255];

  OvcSimpleField1->AsInteger = Form1->NB1->PageIndex;
  OvcCalendar1->CalendarDate = CurrentDate();
  // set picture mask to correspond with the control panel settings
  OvcPictureField1->PictureMask =
    OvcCalendar1->IntlSupport->InternationalDatePChar(Buf, true);
  OvcPictureField1->AsOvcDate = CurrentDate();
  OvcSimpleField2->AsString = "Simple Field Editor";
  OvcPictureField2->AsString = "Picture Field Editor";
  OvcNumericField1->AsFloat = 32768.90;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnAutoClick(TObject *Sender)
{
  static bool Started = false;

  Started = !Started;
  if (Started) {
    T1 = TP->Add(NULL, 1000);
    T2 = TP->Add(NULL, 7000);
    T3 = TP->Add(NULL, 5000);
    btnAuto->Caption = "Stop Auto Demo";
  }
  else {
    TP->RemoveAll();
    btnAuto->Caption = "Start Auto Demo";
  }
}
//---------------------------------------------------------------------------
void __fastcall TForm1::NB1PageChange(TObject *Sender, int Index,
  bool &AllowChange)
{
  OvcSimpleField1->AsInteger = Index;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnCloseClick(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------

void __fastcall TForm1::TPAllTriggers(TObject *Sender, int Handle,
      DWORD Interval, int ElapsedTime)
{
  static int C = 0;

  if (Handle == T1) {
    C++;
    if (C > NB1->PageCollection->Count-1)
      C = 0;
    if (NB1->IsValid(C))
      NB1->PageIndex = C;
  }

  if (Handle == T2) {
    btnTabPosClick(Sender);
  }

  if (Handle == T3) {
    btnTabDisableClick(Sender);
  }
}
//---------------------------------------------------------------------------

File: examples/Delphi/EXNBKU.pas
{$J+} {Writable constants}

unit Exnbku;
    
interface

uses
  SysUtils, WinTypes, WinProcs, Messages, Classes, Graphics, Controls,
  Forms, Dialogs, StdCtrls, OvcNbk, OvcBase, OvcEf, OvcSf,
  OvcCal, OvcPb, OvcPf, OvcNf, OvcMisc, OvcSc, OvcData, ExtCtrls, Buttons,
  OvcTimer, OvcDate;

type
  TForm1 = class(TForm)
    btnTabPos: TButton;
    btnShadowText: TButton;
    Label1: TLabel;
    btnChangeTab: TButton;
    btnFontColor: TButton;
    ColorDialog1: TColorDialog;
    FontDialog1: TFontDialog;
    btnFontSel: TButton;
    NB1: TOvcNotebook;
    OvcSimpleField1: TOvcSimpleField;
    OvcNotebook2: TOvcNotebook;
    OvcCalendar1: TOvcCalendar;
    Label3: TLabel;
    OvcPictureField1: TOvcPictureField;
    btnCalDate: TButton;
    OvcSimpleField2: TOvcSimpleField;
    OvcPictureField2: TOvcPictureField;
    OvcNumericField1: TOvcNumericField;
    Label4: TLabel;
    Label5: TLabel;
    btnTabDisable: TButton;
    OvcSimpleField3: TOvcSimpleField;
    CheckBox1: TCheckBox;
    Edit1: TEdit;
    Edit2: TEdit;
    Label6: TLabel;
    OvcFourWaySpinner2: TOvcSpinner;
    Panel1: TPanel;
    btnClose: TBitBtn;
    OvcController1: TOvcController;
    OvcSpinner1: TOvcSpinner;
    procedure btnTabPosClick(Sender: TObject);
    procedure btnShadowTextClick(Sender: TObject);
    procedure btnChangeTabClick(Sender: TObject);
    procedure btnFontColorClick(Sender: TObject);
    procedure btnFontSelClick(Sender: TObject);
    procedure btnCalDateClick(Sender: TObject);
    procedure OvcCalendar1Change(Sender: TObject; Date: TDateTime);
    procedure btnTabDisableClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure NB1PageChanged(Sender: TObject; Index: Integer);
    procedure btnCloseClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}


procedure TForm1.btnTabPosClick(Sender: TObject);
begin
  with NB1 do
    if TabOrientation = toRight then
      TabOrientation := toTop
    else
      TabOrientation := toRight;
end;

procedure TForm1.btnShadowTextClick(Sender: TObject);
begin
  with NB1 do
    ShadowedText := not ShadowedText;
end;

procedure TForm1.btnChangeTabClick(Sender: TObject);
begin
  NB1.PageIndex := OvcSimpleField1.AsInteger;
end;

procedure TForm1.btnFontColorClick(Sender: TObject);
begin
  ColorDialog1.Color := NB1.Font.Color;
  ColorDialog1.Execute;
  NB1.Font.Color := ColorDialog1.Color;
end;

procedure TForm1.btnFontSelClick(Sender: TObject);
begin
  FontDialog1.Font := NB1.Font;
  FontDialog1.Execute;
  NB1.Font := FontDialog1.Font;
end;

procedure TForm1.btnCalDateClick(Sender: TObject);
begin
  {second page (index 1) contains the calendar}
  NB1.PageIndex := 1;
  {change the calendar date}
  OvcCalendar1.CalendarDate := OvcPictureField1.AsDateTime;
end;

procedure TForm1.OvcCalendar1Change(Sender: TObject; Date: TDateTime);
begin
  Panel1.Caption := OvcCalendar1.DateString('www, dd nnn yyyy');
end;

procedure TForm1.btnTabDisableClick(Sender: TObject);
begin
  with NB1 do
    Pages[3].Enabled := not Pages[3].Enabled;
end;

procedure TForm1.FormCreate(Sender: TObject);
var
  Buf : array[0..255] of Char;
begin
  OvcSimpleField1.AsInteger := Form1.NB1.PageIndex;
  OvcCalendar1.CalendarDate := CurrentDate;
  {set picture mask to correspond with the control panel settings}
  OvcPictureField1.PictureMask :=
    StrPas(OvcCalendar1.IntlSupport.InternationalDatePChar(Buf, True));
  OvcPictureField1.AsOvcDate := CurrentDate;
  OvcSimpleField2.AsString := 'Simple Field Editor';
  OvcPictureField2.AsString := 'Picture Field Editor';
  OvcNumericField1.AsFloat := 32768.90;
end;

procedure TForm1.NB1PageChanged(Sender: TObject; Index: Integer);
begin
  OvcSimpleField1.AsInteger := Index;
end;

procedure TForm1.btnCloseClick(Sender: TObject);
begin
  Close;
end;

end.
